### DISCUSSION TITLE
Forking vs. Referencing

### DISCUSSION THREAD

**Discussion by ahmadabdalla** [Azure/ResourceModules#40] [Q&A] [answered]
If I were a customer. And I'd like to take advantage of this public repo for developing all of my specific templates in my organisation. Constant updates are beneficial to unlock new features from new APIs, but how would I be able to pin certain APIs for resources? I don't want to fork it as then I have to maintain the fork.

What would be our guidance to our customers?

**Comment by segraef** [comment]
It's about applying open source principles to an organization's development work with the help of GitHub.com. See GitHub's white paper [An introduction to innersource.](https://resources.github.com/whitepapers/introduction-to-innersource/)" and the [Open Source Initiative](https://opensource.org/about).

Yes, one of the biggest benefits is to use the power of open source project with the help of forks to get constant updates. Also and most commonly, forks are used to either propose changes to projects like this or to use it as a starting point for the customers organization's idea. You can fork a repository to create a copy of the repository and make changes without affecting the upstream repository.

Another interesting aspect is also to shift the mindset around

"I found an issue, let me create a bug so they can fix it."

But what about instead being a real part of open source and start contributing. You can use forks to propose changes related to fixing a bug. Rather than logging an issue for a bug you've found, you can:

"Fork the repository. Make the fix. Submit a pull request. Fixed."

**Comment by segraef** [comment, marked as answer]
However, I just realized that you were addressing a different and much more important point, referencing.
This will be solved with the help of GitHub Releases.
We started with a pre-release already: https://github.com/Azure/ResourceModules/releases/tag/v0.1-pre
This gives you the opportunity to pin to a specific reposiroty version which holds a specific version of your module.
This might also change or shift with the introduction of the Bicep Registry.

## References
- https://github.com/Azure/bicep/issues/2128
- https://github.com/Azure/bicep/issues/660

**Reply by ahmadabdalla** [reply]
Cool thanks. Let’s close this issue then
